src/crystal/scheduler.cr

```diff
@@ -42,24 +58,30 @@ class Crystal::Scheduler
     Thread.current.scheduler.yield(fiber)
   end
 
+  @worker_in : IO
+  @worker_out : IO
```

Comment:
I saw a commit where these were extracted to a `class` and then that was reverted? Why? Was it because of the overhead of using a `class`? Maybe using `struct` would have worked equally well without the memory overhead?